### PR TITLE
Bionics add weight to character

### PR DESCRIPTION
**Pull request by Fris0uman**
#### Summary
```SUMMARY: Balance "Bionics add weight to character"```

#### Purpose of change
Bionics add weight to character.

#### Describe the solution
Add bionic_weight() function that sums the weight of all installed bionics
Use bionic_weight() in get_weight()

#### Additional context
This could make riding animals and bike difficult/impossible for cyborgs.

I've edited weight of CBMs, it's not super easy to estimate so I'm open to suggestions to adjust them properly

### CODE REVIEW

**Comment by LaVeyanFiend** [comment]
RIP Bionic Cowboy dreams 😢

**Comment by Night-Pryanik** [comment]
With the weight of bionics your provided even a dozen of installed bionics could add only ~5 kilograms which is in fact a nigh negligible mass increase.

File: src/character.cpp [hunk 1]
```diff
@@ -3355,6 +3356,16 @@ units::mass Character::bodyweight() const
     return units::from_kilogram( get_bmi() * pow( height() / 100.0f, 2 ) );
 }
 
+units::mass Character::bionics_weight() const
+{
+    units::mass bio_weight = units::from_gram( 0 );
```
Comment (BevapDin) [inline src/character.cpp]:
```suggestion
    units::mass bio_weight = 0_gram;
```

File: src/character.cpp [hunk 2]
```diff
@@ -3355,6 +3356,16 @@ units::mass Character::bodyweight() const
...
+    for( const auto bio : *my_bionics ) {
+        const item cbm( bio.id.c_str() );
+        bio_weight += cbm.weight();
```
Comment (BevapDin) [inline src/character.cpp]:
```suggestion
        bio_weight += item::find_type( bio.id.c_str() )->weight;
```
No need to create an actual item instance.

File: src/character.h [hunk 3]
```diff
@@ -841,6 +841,8 @@ class Character : public Creature, public visitable<Character>
         int height() const;
         // returns bodyweight of the Character
         units::mass bodyweight() const;
+        // returns total weight of install bionics
```
Comment (BevapDin) [inline src/character.h]: What are "install bionics"?

Comment (Fris0uman) [reply src/character.h]: It's a typo, I meant installed bionics

**Comment by Fris0uman** [comment]
jenkins rebuild

**Comment by dunstad** [comment]
Some bionics replace a significant amount of tissue; one way to account for that could be to make the bionic weight correspondingly lighter. This would be a little weird as the weight of bionics would no longer represent their actual weight, but would stop the player from being weighted down by both bionics and the tissue they replaced.

Probably not an issue for most bionics, but might be worth considering for especially large ones.

**Pull request merged**